node/GUIDE.md

```diff
@@ -0,0 +1,230 @@
+# Setting up a new node service with TChannel, Thrift & Hyperbahn
+
+This guide will show you how to write a server and client.
+
+The code matching this guide is [here](./examples/keyvalue/).
+
+## Dependencies
+
+Make sure you have `node` and `npm` installed. To get started
+create an empty project with `npm init` and run
+`npm install tchannel --save`.
+
+## Defining your service
+
+To define the interface of your service you can create a
+[Thrift](https://thrift.apache.org) file.
+
+Today we will use:
+
+`keyvalue.thrift`:
+```thrift
+struct GetResult {
+    1: string value
+}
+
+service KeyValue {
+    GetResult get_v1(
+        1: string key
+    )
+    void put_v1(
+        1: string key,
+        2: string value
+    )
+}
+```
+
+Our service will have two tchannel endpoints that can be.
+Namely `KeyValue::get_v1` and `KeyValue::put_v1`.
+
+## Writing a tchannel server
+
+To get a server up and running you need to do the following:
+
+1. Create a TChannel and listen on a port
+2. Configure your endpoint handlers using TChannelThrift
+3. Create a hyperbahn client and register with hyperbahn
+
+### Creating a tchannel
+
+Create a tchannel using [`TChannel(opts)`](./docs/channel.md#var-channel--tchanneloptions) and listen to it by calling [`.listen(port, host)`](./docs/channel.md#channellistenport-host-callback).
+
+It's advised that you listen on the external IP of your host to
+ensure that other machines can create incoming connections to you.
+Consider using [`my-local-ip`](https://github.com/dominictarr/my-local-ip) to determine your external IP.
+
+When you create your channel make sure you configure the correct
+options for your use cases. See the [tchannel docs](./docs/channel.md)
+for more details.
+
+```js
+var TChannel = require('tchannel');
+var myLocalIp = require('my-local-ip');
+
+var rootChannel = TChannel();
+rootChannel.listen(4040, myLocalIp());
+```
+
+### Creating and registering your handlers
+
+To register your interface you must implement the `get` and `put`
+endpoints on your tchannel.
+
+First we create a sub channel using [`channel.makeSubChannel()`](./docs/sub-channels.md#channelmakesubchanneloptions).
+
+Once we have a sub channel we have to create a [`TChannelThrift(opts)`](./docs/as-thrift.md#var-tchannelthrift--tchannelthriftops) codec.
+
+Finally you can call [`.register()`](./docs/as-thrift.md#tchannelthriftregistertchannel-arg1-ctx-handlerfn) on the thrift
+codec to register your actual endpoints.
+
+```js
+var TChannelThrift = rootChannel.TChannelAsThrift;
+
+var keyChan = rootChannel.makeSubChannel({
+    serviceName: 'keyvalue'
+});
+var keyThrift = TChannelThrift({
+    source: fs.readFileSync('./keyvalue.thrift', 'utf8')
+});
+var ctx = {
+    store: {}
+};
+
+keyThrift.register(keyChan, 'KeyValue::get_v1', ctx, get);
+keyThrift.register(keyChane, 'KeyValue::put_v1', ctx, put);
+
+function get(context, req, head, body, cb) {
+    cb(null, {
```

Comment:
I'm not sure if this is a Node-ism, but I'm not sure what the "null" parameter for callback is. It might be worth adding a comment to the code example so explain it.

I'm extending my Go guide with this, might be worth doing here too - which is to add an expected exception to the thrift file for if the key is not found, and return that here if the key passed to get is not found).